### PR TITLE
Add StatusOptions to Waku Status packet

### PR DESCRIPTION
**Pull request by kdeme**
Use the association list status-options as in spec: https://github.com/vacp2p/specs/blob/master/waku/waku.md#packet-usage
Also remove anything WakuMode related as this was still leftover from the PoC when compatibility with Whisper was still a thing.

### CODE REVIEW

File: eth/p2p/rlpx_protocols/waku_protocol.nim [hunk 1]
```diff
@@ -116,6 +104,89 @@ type
...
+    except RlpTypeMismatch:
+      # skip unknown keys and their value
+      rlp.skipElem()
+      rlp.skipElem()
+      continue
+
+    case k
+    of powRequirementKey:
+      let pow = rlp.read(uint64)
+      result.powRequirement = some(cast[float64](pow))
```
Comment (zah) [inline eth/p2p/rlpx_protocols/waku_protocol.nim]: https://stackoverflow.com/questions/9056373/what-is-the-difference-between-big-and-little-endian-floats

This source suggests that the binary representation of floats may vary across machines. The Waku protocol should specify what encoding is being used and the source code should probably feature some test cases with sanity checks that the expected bits are being produced.

File: eth/p2p/rlpx_protocols/waku_protocol.nim [hunk 2]
```diff
@@ -175,49 +245,37 @@ p2pProtocol Waku(version = wakuVersion,
...
+    let list = StatusOptions(
+      powRequirement: some(wakuNet.config.powRequirement),
+      bloomFilter: some(wakuNet.config.bloom),
+      lightNode: some(wakuNet.config.isLightNode),
+      confirmationsEnabled: some(wakuNet.config.confirmationsEnabled),
+      rateLimits: some(wakuNet.config.rateLimits),
+      topicInterest: wakuNet.config.topics)
+
+    let m = await peer.status(wakuVersion, list,
+      timeout = chronos.milliseconds(500))
```
Comment (zah) [inline eth/p2p/rlpx_protocols/waku_protocol.nim]: Isn't this timeout rather low for internet usage?

Comment (kdeme) [reply eth/p2p/rlpx_protocols/waku_protocol.nim]: Yes, increased now.

File: eth/p2p/rlpx_protocols/waku_protocol.nim (hunk 1 above)
Comment (kdeme) [reply eth/p2p/rlpx_protocols/waku_protocol.nim]: Good point, we'll have to add this to the spec + and provide a proper call for it per endianness.

**Pull request merged**